src/appmixer/google/gmail/ReplyToEmail/ReplyToEmail.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+const GoogleApi = require('googleapis');
+const commons = require('../../google-commons');
+const mailcomposer = require('mailcomposer');
+const Promise = require('bluebird');
+
+// GoogleApi initialization & promisify of some API functions for convenience
+const gmail = GoogleApi.gmail('v1');
+const send = Promise.promisify(gmail.users.messages.send, { context: gmail.users.messages });
+const modify = Promise.promisify(gmail.users.messages.modify, { context: gmail.users.messages });
+
+/**
+ * GMail send email component.
+ * @extends {Component}
+ */
+module.exports = {
+
+    async receive(context) {
+
+        /**
+         * @type MailComposer
+         */
+        let mail = Object.assign({}, context.messages.in.content);
+        if (mail.from && mail.sender) {
+            mail.from = `${mail.sender} <${mail.from}>`;
+        } else if (mail.sender) {
+            mail.from = `${mail.sender} <${context.profileInfo.email}>`;
+        }
+
+        if (context.messages.in.content.cc) {
```

Comment:
this is redundant, right? 
as Object.assign put the `cc` and `bcc` into the object

also try to simplify, what about
(NOT TESTED)
```js 
 async receive(context) {

        /**
         * @type MailComposer
         */

        const {
            from = context.profileInfo.email,
            sender,
            cc,
            bcc,
            attachments = {},
            thread: threadId,
            labels,
            signature
        } = context.messages.in.content;

        const mail = {
            from: `${sender} <${from}>`,
            cc,
            bcc,
            attachments: await getAttachmentsContent(context, attachments),
            threadId
        };


        // Add signature to the email content if provided
        // TODO v.talas: add more comprehensive comment
        if (signature) {
            if (mail.html) {
                mail.html += `<br><br>${signature}`;
            } else if (mail.text) {
                mail.html = `${mail.text.replace(/\n/g, '<br>')}<br><br>${signature}`;
                delete mail.text;
            } else {
                mail.html = context.messages.in.content.signature;
            }
        }

        return new Promise((resolve, reject) => {
            const composer = mailcomposer(mail);
            composer.keepBcc = true;
            composer.build((err, email) => {
                if (err) {
                    return reject(err);
                }
                resolve(email);
            });
        }).then(async email => {
            // TODO v.talas - weird combination of promises and awaits 
            const result = await send({
                auth: commons.getOauth2Client(context.auth),
                userId: 'me',
                quotaUser: context.auth.userId,
                // URI-safe base64
                resource: {
                    raw: email.toString('base64').replace(/\+/gi, '-').replace(/\//gi, '_'),
                    threadId: mail.threadId
                }
            });

            if (labels && result.id) {
                await modify({
                    auth: commons.getOauth2Client(context.auth),
                    userId: 'me',
                    id: result.id,
                    resource: {
                        addLabelIds: context.messages.in.content.labels.AND.map(label => label.name)
                    }
                });
            } else {
                // TODO v.talas context.SendError is deprecated, throw an error instead 
                await context.sendError('Invalid email label, ' + JSON.stringify(email));
            }

            return context.sendJson(result, 'email');
        });
    }

```